source/exe/signal_action.cc

```diff
@@ -42,21 +42,24 @@ void SignalAction::installSigHandlers() {
   stack.ss_size = altstack_size_;        // ... guard page at the other
   stack.ss_flags = 0;
 
-  RELEASE_ASSERT(sigaltstack(&stack, nullptr) == 0);
+  RELEASE_ASSERT(sigaltstack(&stack, &previous_altstack_) == 0);
 
+  int hidx = 0;
```

Comment:
See the other comment for the thought process that ultimately got me here, combined with a philosophical desire to place the array inside the object rather than behind additional pointers  since the size is known at compile time (this may be a false economy in terms of performance, and almost certainly is irrelevant in this particular use case).  Ultimately I wish for the ability to write a range based for loop across two identically sized iterables but alas this is not part of the language.

Using an iterator type here does require the somewhat awkward " &(*iter) " dance when passing to sigaction and pushes logic and allocations to run time.  I'm not sure that's actually net clearer, but I'm open to being convinced otherwise :)